syne_tune/blackbox_repository/blackbox_surrogate.py

```diff
@@ -208,21 +442,46 @@ def add_surrogate(blackbox: Blackbox, surrogate=None, configuration_space=None):
         The model is fit on top of pipeline that applies basic feature-processing
         to convert rows in X to vectors. We use `config_space` to deduce the types
         of columns in X (categorical parameters are 1-hot encoded).
-    :param configuration_space: configuration space for the resulting blackbox surrogate.
-        The default is `blackbox.configuration_space`. But note that if `blackbox`
-        is tabular, the domains in `blackbox.configuration_space` are typically
-        categorical even for numerical parameters.
+    :param configuration_space: configuration space for the resulting blackbox
+        surrogate. The default is `blackbox.configuration_space`. But note that
+        if `blackbox` is tabular, the domains in `blackbox.configuration_space`
+        are typically categorical even for numerical parameters.
+    :param predict_curves: If True, the surrogate uses multivariate regression
+        to predict metric curves over fidelities. If False, fidelity is used
+        as input. The latter can lead to inconsistent predictions along
+        fidelity and is typically more expensive
+    :param separate_seeds: If True, seeds in `blackbox` map to seeds in the
+        surrogate blackbox, which fits different models to each seed. If False,
+        the data from `blackbox` is merged for all seeds, and the surrogate
+        represents a single seed. The latter provides more data for the surrogate
+        model to be fit, but the variation between seeds is lost in the
+        surrogate
+    :param fit_differences: Names of objectives which are cumulative sums. For
+        these objectives, the `y` data is transformed to finite differences
+        before fitting the model. This is recommended for `elapsed_time`
+        objectives.
     :return: a blackbox where the output is obtained through the fitted surrogate
     """
-    surrogate = _default_surrogate(surrogate)
     if configuration_space is None:
         configuration_space = blackbox.configuration_space
-    X, y = blackbox.hyperparameter_objectives_values()
+    if separate_seeds and blackbox.fidelity_values is not None:
+        num_seeds = len(blackbox.fidelity_values)
+    else:
+        num_seeds = 1
+    if predict_curves is None:
+        # Default is True for `BlackboxTabular`, False for
+        # `BlackboxOffline` (which does not support True right now)
+        predict_curves = not isinstance(blackbox, BlackboxOffline)
```

Comment:
this default value is hidden and not documented

Reply:
OK